src/api/v1/utils.ts

```diff
@@ -5,6 +5,28 @@ import * as Hapi from 'hapi';
 import * as Boom from 'boom';
 import { ApiResponse } from './schema/response';
 
+export interface JoiBoomError extends Boom {
+  details: {message: string}[];
+}
+
+/*
+ * formatJoiValidation allows joi errors thrown from
+ * failAction to be reformatted before sending response.
+ * Note joi throws an error on the first value that fails
+ * validation, therefore 'details' will only have 1 element.
+ */
+
+export const formatJoiValidation = ({ output: { payload }, details }: JoiBoomError)
```

Comment:
I think this function can probably be folded into the other one once we use declaration merging  as in the other comment.